libraries/helper.rb

```diff
@@ -46,22 +46,44 @@ def omnibus_env
       node.run_state[:omnibus_env] ||= Hash.new { |hash, key| hash[key] = [] } # ~FC001
     end
 
-    def omnibus_toolchain_enabled?
-      # We don't have an omnibus toolchain for windows yet.
-      !windows?
+    def toolchain_install_dir
+      if windows?
+        windows_safe_path_join(ENV['SYSTEMDRIVE'], 'opscode', node['omnibus']['toolchain_name'])
+      else
+        "/opt/#{node['omnibus']['toolchain_name']}"
+      end
     end
 
     def build_user_shell
-      if omnibus_toolchain_enabled?
-        "/opt/#{node['omnibus']['toolchain_name']}/bin/bash"
+      if windows?
+        windows_safe_path_join(toolchain_install_dir, 'embedded', 'bin', 'usr', 'bin', 'bash')
       else
-        '/usr/local/bin/bash'
+        ::File.join(toolchain_install_dir, 'bin', 'bash')
       end
     end
 
     def windows_arch_i386?
       windows? && (i386? || (node.name =~ /i386/))
     end
+
+    def mingw_toolchain_name
+      windows_arch_i386? ? 'mingw32' : 'mingw64'
+    end
+
+    def msys2_path(path)
+      path.sub(%r{^([A-Za-z]):/}, '/\\1/')
```

Comment:
So the behavior looks equivalent but either version of this regex doesn't appear to do anything for paths with backward slashes...not sure if it should:

```irb
irb(main):026:0> forward_path = 'C:/foo/bar'
=> "C:/foo/bar"
irb(main):027:0> backward_path = 'C:\foo\bar'
=> "C:\\foo\\bar"
irb(main):028:0> forward_path.sub(/^([A-Za-z]):\//, "/\\1/")
=> "/C/foo/bar"
irb(main):029:0> backward_path.sub(/^([A-Za-z]):\//, "/\\1/")
=> "C:\\foo\\bar"
irb(main):030:0> forward_path.sub(%r{^([A-Za-z]):/}, '/\\1/')
=> "/C/foo/bar"
irb(main):031:0> backward_path.sub(%r{^([A-Za-z]):/}, '/\\1/')
=> "C:\\foo\\bar"
```

You wrote the original `msys2_path` function @tduffield, was your intention for either version of a Windows path to have the `:` removed?